### PR TITLE
feat : create page UI prototype

### PR DESCRIPTION
**Pull request by jzongyun**
## Docs

- [figma design template](https://www.figma.com/file/0ByGSnJpm6DdAwqOrPSSeR/Untitled?type=design&mode=design&t=YzW6JA1YS0Tq0IU0-1)

## Changes

- before :
- after : 스플레시 페이지, 캘린더 페이지, 일기작성 페이지, 통계 페이지의 프로토타입 구현
- images

## Test Checklist

- [x] check 1 모든 페이지가 정상적으로 작동하는지 확인

### CODE REVIEW

File: MangoDiary/public/images/logo2.png
Comment (kim-song-jun) [inline MangoDiary/public/images/logo2.png:1]: change a meaningful name

File: MangoDiary/public/images/mood1.jpg
Comment (kim-song-jun) [inline MangoDiary/public/images/mood1.jpg:1]: change a meaningful name

File: MangoDiary/public/images/mood2.jpg
Comment (kim-song-jun) [inline MangoDiary/public/images/mood2.jpg:1]: change a meaningful name

File: MangoDiary/public/images/mood2c.jpg
Comment (kim-song-jun) [inline MangoDiary/public/images/mood2c.jpg:1]: change a meaningful name

File: MangoDiary/public/images/mood3.jpg
Comment (kim-song-jun) [inline MangoDiary/public/images/mood3.jpg:1]: change a meaningful name

File: MangoDiary/src/components/CalendarPage.vue [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+      </div>
+      <div>
+          <select v-model="selectedYear" @change="updateCalendar">
+
+            <option v-for="year in yearRange" :key="year" :value="year">{{ year }}</option>
+          </select>
+        </div>
+        <div>
+          <select v-model="selectedMonth" @change="updateCalendar">
+
```
Comment (kim-song-jun) [inline MangoDiary/src/components/CalendarPage.vue]: delete line

File: MangoDiary/src/components/CalendarPage.vue [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+            <option v-for="(month, index) in 12" :key="index + 1" :value="index + 1">{{ monthNames[index] }}</option>
```
Comment (kim-song-jun) [inline MangoDiary/src/components/CalendarPage.vue]: key must be immutable value

File: MangoDiary/src/components/CalendarPage.vue [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+          </select>
+        </div>
+    </header>
+
```
Comment (kim-song-jun) [inline MangoDiary/src/components/CalendarPage.vue:18]: delete empty line

File: MangoDiary/src/components/CalendarPage.vue [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+    <div class="week">
+      <div class="day-header" v-for="dayName in dayNames" :key="dayName">{{ dayName }}</div>
+    </div>
+
+
+  <div class="calendar">
+      <div class="week" v-for="week in calendar" :key="week">
+          <div class="day" v-for="day in week" :key="day">
+              {{ day == null ? "-" : day }}
```
Comment (kim-song-jun) [inline MangoDiary/src/components/CalendarPage.vue]: make display function like this
```
showDay(day) {
  return day == null ? '-' : day
}
```

File: MangoDiary/src/components/CalendarPage.vue [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+              <div class="day-container" :class="{ today: isToday(day) }" @click="writeDiary()"></div>
```
Comment (kim-song-jun) [inline MangoDiary/src/components/CalendarPage.vue]: check css parameter

File: MangoDiary/src/components/CalendarPage.vue [hunk 6]
```diff
@@ -0,0 +1,157 @@
...
+    </footer>
+  </div>
+
+</template>
+
+<script>
+export default {
+  data() {
+    return {
+      selectedYear: new Date().getFullYear(),
```
Comment (kim-song-jun) [inline MangoDiary/src/components/CalendarPage.vue]: make function like this
```
getToday() {
  this.selectedYear: new Date().getFullYear()
   ...
}

mounted() {
   this.getToday()
}
```

File: MangoDiary/src/components/CalendarPage.vue [hunk 7]
```diff
@@ -0,0 +1,157 @@
...
+      selectedMonth: new Date().getMonth() + 1,
+      today: new Date().getDate(),
+      dayNames: ['Sun', 'Mon', 'Tues', 'Wed', 'Thur', 'Fri', 'Sat'],
```
Comment (kim-song-jun) [inline MangoDiary/src/components/CalendarPage.vue]: how about using vuex?

File: MangoDiary/src/components/CalendarPage.vue [hunk 8]
```diff
@@ -0,0 +1,157 @@
...
+    };
+  },
+  computed: {
+    yearRange() {
+        const currentYear = new Date().getFullYear();
+        return Array.from({ length: 10 }, (_, index) => currentYear - index);
+    },
+    monthNames() {
```
Comment (kim-song-jun) [inline MangoDiary/src/components/CalendarPage.vue]: see above

File: MangoDiary/src/components/CalendarPage.vue [hunk 9]
```diff
@@ -0,0 +1,157 @@
...
+    calendar() {
+      return this.generateCalendar();
+    },
+  },
+  methods: {
+    generateCalendar() {
+      const firstDay = new Date(this.selectedYear, this.selectedMonth - 1, 1).getDay();
+      const daysInMonth = new Date(this.selectedYear, this.selectedMonth, 0).getDate();
+
+      let calendar = [];
```
Comment (kim-song-jun) [inline MangoDiary/src/components/CalendarPage.vue]: using `const`

File: MangoDiary/src/components/CalendarPage.vue [hunk 10]
```diff
@@ -0,0 +1,157 @@
...
+      let dayCount = 1;
+
+      for (let i = 0; i < 6; i++) {
```
Comment (kim-song-jun) [inline MangoDiary/src/components/CalendarPage.vue]: make const value for calendar `row` and `col`

File: MangoDiary/src/components/CalendarPage.vue [hunk 11]
```diff
@@ -0,0 +1,157 @@
...
+      return calendar;
+    },
+    updateCalendar() {
+      this.calendar = this.generateCalendar();
+    },
+    isToday(day) {
+      return day == this.today;
+    },
+    writeDiary() {
+      this.$router.push('/writing');
```
Comment (kim-song-jun) [inline MangoDiary/src/components/CalendarPage.vue]: path must be handle in variable.

File: MangoDiary/src/components/CalendarPage.vue [hunk 12]
```diff
@@ -0,0 +1,157 @@
...
+.today {
+  background-color: #3498db;
+  color: #fff;
+}
+
+.hasDiary {
+  border: 1px solid #ff9f9f;
+}
+
+footer{
```
Comment (kim-song-jun) [inline MangoDiary/src/components/CalendarPage.vue]: do not use tag selector. make some css class

File: MangoDiary/src/components/SplashPage.vue [hunk 13]
```diff
@@ -0,0 +1,37 @@
+<template>
+  <div class="splash-page">
+    <img src="/images/logo.png" class="logo">
+    <h4 @click="redirectToCalendar" class="start">Touch to Start</h4>
```
Comment (kim-song-jun) [inline MangoDiary/src/components/SplashPage.vue]: change `redirectToCalendar` to `redirectToCalendar()`

File: MangoDiary/src/components/SplashPage.vue [hunk 14]
```diff
@@ -0,0 +1,37 @@
...
+  </div>
+</template>
+
+<script>
+export default {
+  methods: {
+    redirectToCalendar() {
+      this.$router.push('/calendar');
```
Comment (kim-song-jun) [inline MangoDiary/src/components/SplashPage.vue]: see above

File: MangoDiary/src/components/StatisticsPage.vue [hunk 15]
```diff
@@ -0,0 +1,150 @@
+<template>
+  <div class="statistics-page">
+    <header>
+      <img src="/images/logo2.png" class="logo">
+      <div>
+        <select id="yearSelect" v-model="selectedYear" @change="updateStatistics">
```
Comment (kim-song-jun) [inline MangoDiary/src/components/StatisticsPage.vue]: see above

File: MangoDiary/src/components/StatisticsPage.vue [hunk 16]
```diff
@@ -0,0 +1,150 @@
+<template>
+  <div class="statistics-page">
+    <header>
+      <img src="/images/logo2.png" class="logo">
```
Comment (kim-song-jun) [inline MangoDiary/src/components/StatisticsPage.vue]: see above

File: MangoDiary/src/components/StatisticsPage.vue [hunk 17]
```diff
@@ -0,0 +1,150 @@
...
+      <div>
+        <select id="yearSelect" v-model="selectedYear" @change="updateStatistics">
+          <option v-for="year in yearRange" :key="year" :value="year">{{ year }}</option>
+        </select>
+      </div>
+      <div>
+        <select id="monthSelect" v-model="selectedMonth" @change="updateStatistics">
+          <option v-for="(month, index) in 12" :key="index + 1" :value="index + 1">{{ monthNames[index] }}</option>
```
Comment (kim-song-jun) [inline MangoDiary/src/components/StatisticsPage.vue]: see above

File: MangoDiary/src/components/StatisticsPage.vue [hunk 18]
```diff
@@ -0,0 +1,150 @@
...
+            <td>3</td>
+            <td>
+                <img src="/images/mood5.jpg" class="selectedImage">
+            </td>
+            <td class="emojiDetails">슬픈 날 : {{ 0 }}일</td>
+          </tr>
+      </table>
+    </div>
+    <footer>
+      <router-link to="/calendar">캘린더</router-link>
```
Comment (kim-song-jun) [inline MangoDiary/src/components/StatisticsPage.vue]: check path

File: MangoDiary/src/components/StatisticsPage.vue [hunk 19]
```diff
@@ -0,0 +1,150 @@
...
+  height: 844px;
+  width: 390px;
+}
+.logo {
+  margin-top: 16px;
+  margin-bottom: 25px;
+  width: 100px;
+  height: auto;
+}
+.mostSelectedEmoji{
```
Comment (kim-song-jun) [inline MangoDiary/src/components/StatisticsPage.vue]: change name
`.most-select-emoji`

File: MangoDiary/src/components/StatisticsPage.vue [hunk 20]
```diff
@@ -0,0 +1,150 @@
...
+.emojiDetails{
+  text-align: left;
+  padding-left: 15px;
+  font-size: 15px;
+}
+.selectedImage{
+  width: 70px;
+  height: auto;
+}
+h4 {
```
Comment (kim-song-jun) [inline MangoDiary/src/components/StatisticsPage.vue]: do not use tag selector

File: MangoDiary/src/components/WritePage.vue [hunk 21]
```diff
@@ -0,0 +1,151 @@
...
+          <img v-for="(emoji, index) in moodEmojis" :key="index" :src="emoji" @click="selectEmoji(index)" class="moodList">
+        </div>
+      </div>
+
+      <h3>오늘의 일기</h3>
+      <h4>{{ year }}년 {{ month }}월 {{ day }}일</h4>
+      <textarea v-model="diaryContent" placeholder="오늘 하루는 무엇을 했나요?"></textarea>
+
+      <label for="imageInput">
+        <img class="selctedImage" v-if="!selectedImage" src="/images/photo.png">
```
Comment (kim-song-jun) [inline MangoDiary/src/components/WritePage.vue]: change image to `<div>` box or `input` box

File: MangoDiary/src/components/WritePage.vue [hunk 22]
```diff
@@ -0,0 +1,151 @@
...
+  </div>
+</template>
+
+<script>
+export default {
+  data() {
+    return {
+      diaryContent: '',
+      selectedImage: null,
+      moodEmojis: [
```
Comment (kim-song-jun) [inline MangoDiary/src/components/WritePage.vue]: using vuex

File: MangoDiary/src/components/WritePage.vue [hunk 23]
```diff
@@ -0,0 +1,151 @@
...
+  width: 350px;
+  height: 250px;
+}
+
+header{
+  display: flex;
+  justify-content: space-evenly;
+}
+
+h3 {
```
Comment (kim-song-jun) [inline MangoDiary/src/components/WritePage.vue]: see above

File: MangoDiary/src/main.js [hunk 24]
```diff
@@ -0,0 +1,6 @@
+import { createApp } from 'vue'
+import App from './App.vue'
+
+import router from './router'
```
Comment (kim-song-jun) [inline MangoDiary/src/main.js]: add `.js`

**Review by kim-song-jun — CHANGES_REQUESTED** [review]
👍 
Plz add Footer.

File: MangoDiary/public/images/logo2.png
Comment (kim-song-jun) [reply MangoDiary/public/images/logo2.png:1]: does it changed?

**Comment by jzongyun** [comment]
Done

File: MangoDiary/src/components/CalendarPage.vue (hunk 5 above)
Comment (kim-song-jun) [reply MangoDiary/src/components/CalendarPage.vue]: Is it solved?

File: MangoDiary/src/components/CalendarPage.vue (hunk 9 above)
Comment (kim-song-jun) [reply MangoDiary/src/components/CalendarPage.vue]: Is it solved?

File: MangoDiary/src/components/SplashPage.vue [hunk 25]
```diff
@@ -0,0 +1,35 @@
...
+    <img src="/images/title.png" class="logo">
+    <h4 @click="redirectToCalendar()" class="start">Touch to Start</h4>
+  </div>
+</template>
+
+<script>
+export default {
+  methods: {
+    redirectToCalendar() {
+      this.$router.push(this.$store.state.calendar);
```
Comment (kim-song-jun) [inline MangoDiary/src/components/SplashPage.vue]: delete `;`

File: MangoDiary/src/components/CalendarPage.vue [hunk 26]
```diff
@@ -0,0 +1,153 @@
...
+      let calendar = [];
+      let dayCount = 1;
+
+      for (let i = 0; i < col; i++) {
+        let week = [];
+        for (let j = 0; j < row; j++) {
+          if ((i == 0 && j < firstDay) || (dayCount > daysInMonth)) {
+            week.push(null);
+          } else {
+            week.push(dayCount++);
```
Comment (kim-song-jun) [inline MangoDiary/src/components/CalendarPage.vue]: delete all `;`

File: MangoDiary/src/components/StatisticsPage.vue [hunk 27]
```diff
@@ -0,0 +1,157 @@
...
+<script>
+export default {
+  data() {
+    return {
+    };
+  },
+  computed: {
+    yearRange() {
+      const currentYear = new Date().getFullYear();
+      return Array.from({ length: 10 }, (_, index) => currentYear - index);
```
Comment (kim-song-jun) [inline MangoDiary/src/components/StatisticsPage.vue]: delete `;`

File: MangoDiary/src/components/StatisticsPage.vue [hunk 28]
```diff
@@ -0,0 +1,157 @@
...
+    </div>
+    <footer class="menu-bar">
+      <div @click="goToCalendar">캘린더</div>
+    </footer>
+  </div>
+</template>
+
+<script>
+export default {
+  data() {
```
Comment (kim-song-jun) [inline MangoDiary/src/components/StatisticsPage.vue]: delete `data()`

File: MangoDiary/src/components/WritePage.vue [hunk 29]
```diff
@@ -0,0 +1,161 @@
...
+          <label for="imageInput">+</label>
+          <label>Upload Image</label>
+        </div>
+        <img class="selected-image" v-else :src="selectedImage">
+      </label>
+      <input
+        id="imageInput"
+        type="file"
+        style="display: none"
+        @change="handleImageUpload"
```
Comment (kim-song-jun) [inline MangoDiary/src/components/WritePage.vue]: add `()` end of function.

File: MangoDiary/src/components/WritePage.vue [hunk 30]
```diff
@@ -0,0 +1,161 @@
...
+export default {
+  data() {
+    return {
+      diaryContent: '',
+      selectedImage: null,
+    };
+  },
+  methods: {
+    goToCalendar() {
+      this.$router.push(this.$store.state.calendar);
```
Comment (kim-song-jun) [inline MangoDiary/src/components/WritePage.vue]: remove `;`

File: MangoDiary/src/router.js [hunk 31]
```diff
@@ -0,0 +1,19 @@
+import { createWebHistory, createRouter } from "vue-router";
+
+const routes = [
+    {path: "/",
```
Comment (kim-song-jun) [inline MangoDiary/src/router.js]: add space after `{`

File: MangoDiary/src/store.js [hunk 32]
```diff
@@ -0,0 +1,29 @@
...
+        return {
+            write : '/write',
+            statistics : '/statistics',
+            calendar : '/calendar',
+            selectedYear: 0,
+            selectedMonth: 0,
+            today: 0,
+            dayNames: ['Sun', 'Mon', 'Tues', 'Wed', 'Thur', 'Fri', 'Sat'],
+            monthNames: ['Jan', 'Feb', 'Mar', 'Apr', 'May', 'June', 'July', 'Aug', 'Sept', 'Oct', 'Nov', 'Dec'],
+            moodEmojis: ['grey/angry','grey/depressed','grey/pleased','grey/happy','grey/sad'],
```
Comment (kim-song-jun) [inline MangoDiary/src/store.js]: how about make prefix for `grey` or `color`?

File: MangoDiary/src/store.js [hunk 33]
```diff
@@ -0,0 +1,29 @@
...
+            originalEmojis: ['grey/angry','grey/depressed','grey/pleased','grey/happy','grey/sad'],
```
Comment (kim-song-jun) [inline MangoDiary/src/store.js]: Why that 2 variable is same? Where it used? Why it needed?

**Review by kim-song-jun — COMMENTED** [review]
🥇

**Review by kim-song-jun — APPROVED** [review]

**Review by kim-song-jun — COMMENTED** [review]
👍

**Pull request merged**